scripts/export-signupless-reportbacks-to-rogue.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Script to export Reportbacks that do not have a corresponding Signup from drupal into Rogue.
+ *
+ * to run:
+ * drush --script-path=../scripts/ php-script export-signupless-reportbacks-to-rogue.php
+ */
+
+// Get all the signupless reportbacks
+$reportbacks = db_query("SELECT rb.rbid, rb.nid, rb.run_nid, rb.quantity, rb.why_participated, rb.rbid, rb.flagged, rb.uid, rb.created
+  FROM dosomething.dosomething_reportback rb
+  LEFT JOIN dosomething.dosomething_signup ds on rb.uid = ds.uid AND rb.run_nid = ds.run_nid
+  WHERE ds.sid IS NULL
+  AND rb.rbid NOT IN (SELECT rbid from dosomething_rogue_reportbacks)");
+
+$client = dosomething_rogue_client();
+
+if (!$reportbacks) {
+  echo 'Nothing to migrate!';
+}
+
+// For each reportback, send a request for each reportback file to POST /posts and Rogue will automatically create the signup
+foreach ($reportbacks as $reportback) {
+  echo 'On rbid ' . $reportback->rbid . '...' . PHP_EOL;
+  $sent_at = date('Y-m-d H:i:s');
+
+  // Try to get Northstar ID
+  $northstar_id = dosomething_user_get_northstar_id($reportback->uid, 'drupal_id');
+
+  if (!$northstar_id) {
+    echo "\t" . 'No northstar id, that is terrible ' . $reportback->rbid . PHP_EOL;
+
+    continue;
+  }
+
+  // Get reportback files
+  $photos = db_query('SELECT rbf.fid, rbf.remote_addr, rbf.caption, rbf.status, rbf.reviewed, rbf.reviewer, rbf.source, rbf.status, rblog.timestamp
+                          FROM dosomething_reportback_file rbf
+                          INNER JOIN dosomething_reportback_log rblog ON rbf.fid = substring_index(rblog.files, \',\',-1)
+                          WHERE rbf.rbid = '. $reportback->rbid . '
+                          AND rbf.fid NOT IN (SELECT fid from dosomething_rogue_reportbacks)
+                          GROUP BY rbf.fid')->fetchAll();
+
+  // Just send a signup if there are no photos (there are about 25 of these)
```

Comment:
I think I might be a little confused on this piece. I don't see where you are looking through the RBS with out photos

Reply:
@sbsmith86 the query above gets all the reportback files associated with the reportback. There are a few cases where that query won't return anything. In those cases, below (`if (!photos)`), we send over that data as just a signup.